File: awiki/python-client.backup.20260308/scripts/capture_e2ee_init.py
#!/usr/bin/env python3
"""Capture and display the actual e2ee_init packet sent to awiki.ai."""

import json
import asyncio
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent))

from utils.config import SDKConfig
from credential_store import load_identity, create_authenticator
from utils.e2ee import E2eeClient
from utils.auth import generate_wba_auth_header
from utils.identity import DIDIdentity

# Target: NodeAgent3
TARGET_DID = "did:wba:awiki.ai:user:k1_136mytd3v6udtuW5ZsfyP7q1hD3iAEsZysJ8xI9nXsY"

print("=" * 80)
print("Python E2EE Init Packet Capture")
print("=" * 80)

# Load identity
cred = load_identity("pythonagent")
if cred is None:
    print("Error: Credential 'pythonagent' not found")
    sys.exit(1)

print(f"\nLocal DID: {cred['did']}")
print(f"Target DID: {TARGET_DID}")

# Create identity object
identity = DIDIdentity(
    did=cred["did"],
    did_document=cred["did_document"],
    private_key_pem=cred["private_key_pem"].encode("utf-8"),
    public_key_pem=cred["public_key_pem"].encode("utf-8"),
    e2ee_signing_private_pem=cred.get("e2ee_signing_private_pem", b"") or None,
    e2ee_agreement_private_pem=cred.get("e2ee_agreement_private_pem", b"") or None,
)

# Create E2EE client
e2ee_client = E2eeClient(
    local_did=cred["did"],
    signing_pem=cred.get("e2ee_signing_private_pem"),
    x25519_pem=cred.get("e2ee_agreement_private_pem"),
)

# Generate e2ee_init
async def capture_init():
    msg_type, content = await e2ee_client.initiate_handshake(TARGET_DID)
    return msg_type, content

msg_type, content = asyncio.run(capture_init())

print(f"\n[Packet 1: e2ee_init]")
print("-" * 80)

# Build the full RPC request
rpc_request = {
    "jsonrpc": "2.0",
    "method": "send",
    "params": {
        "sender_did": cred["did"],
        "receiver_did": TARGET_DID,
        "content": content,
        "type": msg_type
    },
    "id": 1
}

print("\nFull RPC Request:")
print(json.dumps(rpc_request, indent=2))

print("\n[Key Fields Analysis]")
print("-" * 80)
print(f"session_id: {content.get('session_id')}")
print(f"hpke_suite: {content.get('hpke_suite')}")
print(f"enc length: {len(content.get('enc', ''))} chars (base64)")
print(f"encrypted_seed length: {len(content.get('encrypted_seed', ''))} chars (base64)")
print(f"proof.proof_value length: {len(content.get('proof', {}).get('proof_value', ''))} chars (base64url)")

# Decode and show raw bytes
import base64
enc_bytes = base64.b64decode(content['enc'])
encrypted_seed_bytes = base64.b64decode(content['encrypted_seed'])
proof_value_bytes = base64.urlsafe_b64decode(content['proof']['proof_value'] + '=' * (-len(content['proof']['proof_value']) % 4))

print(f"\n[Raw Byte Lengths]")
print(f"enc (X25519 ephemeral public key): {len(enc_bytes)} bytes")
print(f"encrypted_seed (HPKE ciphertext): {len(encrypted_seed_bytes)} bytes")
print(f"proof_value (R||S signature): {len(proof_value_bytes)} bytes")

print("\n" + "=" * 80)

File: awiki/MIGRATION-proj/nodejs-work/tests/capture_e2ee_init.js
/**
 * Capture and display the actual e2ee_init packet sent to awiki.ai.
 */

import { loadIdentity } from '../src/credential_store.js';
import { E2eeClient } from '../src/e2ee.js';
import { resolveToDid } from '../src/utils/resolve.js';
import { createSDKConfig } from '../src/utils/config.js';

const TARGET_DID = 'did:wba:awiki.ai:user:k1_136mytd3v6udtuW5ZsfyP7q1hD3iAEsZysJ8xI9nXsY';

console.log('='.repeat(80));
console.log('Node.js E2EE Init Packet Capture');
console.log('='.repeat(80));

// Load identity
const cred = loadIdentity('nodeagent3');
if (!cred) {
    console.log('Error: Credential \'nodeagent3\' not found');
    process.exit(1);
}

console.log(`\nLocal DID: ${cred.did}`);
console.log(`Target DID: ${TARGET_DID}`);

// Create E2EE client
const e2eeClient = new E2eeClient(cred.did, {
    signingPem: Buffer.from(cred.e2ee_signing_private_pem, 'utf-8'),
    x25519Pem: Buffer.from(cred.e2ee_agreement_private_pem, 'utf-8')
});

// Generate e2ee_init
async function captureInit() {
    // Resolve peer DID document and extract X25519 public key
    const config = createSDKConfig();
    const peerDoc = await resolveToDid(TARGET_DID, config);
    
    // Extract X25519 public key from peer DID document
    const keyAgreement = peerDoc.keyAgreement?.[0] || peerDoc.did_document?.keyAgreement?.[0];
    const keyId = keyAgreement?.split('#')[1] || 'key-3';
    
    const peerVm = peerDoc.verificationMethod?.find(vm => vm.id.endsWith('#key-3')) ||
                   peerDoc.did_document?.verificationMethod?.find(vm => vm.id.endsWith('#key-3'));
    
    if (!peerVm || !peerVm.publicKeyMultibase) {
        throw new Error('Peer has no X25519 key for E2EE');
    }
    
    // Decode base58btc public key
    const { base58btc } = await import('multiformats/bases/base58');
    const peerPkBytes = base58btc.decode(peerVm.publicKeyMultibase);
    const peerPk = Buffer.from(peerPkBytes);
    
    const [msgType, content] = await e2eeClient.initiateHandshake(TARGET_DID, peerPk, `did:wba:awiki.ai:user:k1_136mytd3v6udtuW5ZsfyP7q1hD3iAEsZysJ8xI9nXsY#${keyId}`);
    return [msgType, content];
}

const [msgType, content] = await captureInit();

console.log(`\n[Packet 1: e2ee_init]`);
console.log('-'.repeat(80));

// Build the full RPC request
const rpcRequest = {
    jsonrpc: '2.0',
    method: 'send',
    params: {
        sender_did: cred.did,
        receiver_did: TARGET_DID,
        content: content,
        type: msgType
    },
    id: 1
};

console.log('\nFull RPC Request:');
console.log(JSON.stringify(rpcRequest, null, 2));

console.log('\n[Key Fields Analysis]');
console.log('-'.repeat(80));
console.log(`session_id: ${content.session_id}`);
console.log(`hpke_suite: ${content.hpke_suite}`);
console.log(`enc length: ${content.enc?.length || 0} chars (base64)`);
console.log(`encrypted_seed length: ${content.encrypted_seed?.length || 0} chars (base64)`);
console.log(`proof.proof_value length: ${content.proof?.proof_value?.length || 0} chars (base64url)`);

// Decode and show raw bytes
import { Buffer } from 'buffer';

const encBytes = Buffer.from(content.enc, 'base64');
const encryptedSeedBytes = Buffer.from(content.encrypted_seed, 'base64');
const proofValueBytes = Buffer.from(content.proof.proof_value, 'base64url');

console.log('\n[Raw Byte Lengths]');
console.log(`enc (X25519 ephemeral public key): ${encBytes.length} bytes`);
console.log(`encrypted_seed (HPKE ciphertext): ${encryptedSeedBytes.length} bytes`);
console.log(`proof_value (R||S signature): ${proofValueBytes.length} bytes`);

console.log('\n' + '='.repeat(80));
